eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/inmemory/AsyncInMemoryEventStorageEngine.java

```diff
@@ -169,7 +169,7 @@ public MessageStream<EventMessage<?>> source(@Nonnull SourcingCondition conditio
         }
 
         return eventsToMessageStream(condition.start(),
-                                     Math.min(condition.end(), eventStorage.lastKey()),
+                                     eventStorage.isEmpty() ? -1 : Math.min(condition.end(), eventStorage.lastKey()),
```

Comment:
`eventStorage.lastKey()` was throwing exception if we'd like to source from empty EventStorageEngine.